test/minikube/minikube_domain_resync_test.go

```diff
@@ -0,0 +1,193 @@
+// +build enterprise
+
+package minikube
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/nuodb/nuodb-helm-charts/test/testlib"
+
+	"gotest.tools/assert"
+
+	"github.com/gruntwork-io/terratest/modules/helm"
+	"github.com/gruntwork-io/terratest/modules/k8s"
+)
+
+func verifyProcessLabels(t *testing.T, namespaceName string, adminPod string) (archiveVolumeClaims map[string]string) {
+	options := k8s.NewKubectlOptions("", "")
+	options.Namespace = namespaceName
+
+	output, err := k8s.RunKubectlAndGetOutputE(t, options, "exec", adminPod, "--",
+		"nuocmd", "--show-json", "get", "processes", "--db-name", "demo")
+	assert.NilError(t, err, output)
+
+	err, objects := testlib.Unmarshal(output)
+	assert.NilError(t, err, output)
+
+	archiveVolumeClaims = make(map[string]string)
+	for _, obj := range objects {
+		podName, ok := obj.Labels["pod-name"]
+		assert.Check(t, ok)
+		// check that Pod exists
+		pod := k8s.GetPod(t, options, podName)
+
+		containerId, ok := obj.Labels["container-id"]
+		assert.Check(t, ok)
+		// check that Pod has container ID
+		for _, containerStatus := range pod.Status.ContainerStatuses {
+			assert.Equal(t, "docker://"+containerId, containerStatus.ContainerID)
+		}
+
+		claimName, ok := obj.Labels["archive-pvc"]
+		if ok {
+			assert.Equal(t, "SM", obj.Type, "archive-pvc label should only be present for SMs")
+			// check that PVC exists
+			k8s.RunKubectl(t, options, "get", "pvc", claimName)
+			// add mapping of PVC to archive ID
+			output, err = k8s.RunKubectlAndGetOutputE(t, options, "exec", adminPod, "--",
+				"nuocmd", "get", "value", "--key", "archiveVolumeClaims/"+claimName)
+			assert.NilError(t, err)
+			archiveVolumeClaims[claimName] = strings.TrimSpace(output)
+		} else {
+			assert.Equal(t, "TE", obj.Type, "archive-pvc label should only be absent for TEs")
+		}
+	}
+	return archiveVolumeClaims
+}
+
+type archive struct {
+	ArchiveId string `json:"archiveId"`
+	DbName    string `json:"dbName"`
+}
+
+func unmarshalArchives(s string) (err error, processes []archive) {
+	dec := json.NewDecoder(strings.NewReader(s))
+
+	for {
+		var obj archive
+		err = dec.Decode(&obj)
+		if err == io.EOF {
+			// all done
+			return nil, processes
+		}
+
+		if err != nil {
+			return
+		}
+
+		processes = append(processes, obj)
+	}
+}
```

Comment:
I would break this out into its own class. Same as the other JSON/YAML to Go parsers

Reply:
I swiped this from the existing `Unmarshal` function, hence the bogus `processes` variable name. You're saying promote this so that it can be used elsewhere?